src/Drips.sol

```diff
@@ -520,40 +501,36 @@ library Drips {
         uint32 updateTime,
         uint32 defaultEnd
     ) private view returns (uint32 start, uint32 end) {
-        return _dripsRange(receiver, updateTime, defaultEnd, 0, _currTimestamp());
+        return _dripsRange(receiver, defaultEnd, updateTime, _currTimestamp());
     }
 
     /// @notice Calculates the time range in the future in which a receiver will be dripped to.
     /// @param receiver The drips receiver
-    /// @param updateTime The time when drips are configured
     /// @param defaultEnd The end time of drips without duration
-    function _dripsRangeInFuture(
-        DripsReceiver memory receiver,
-        uint32 updateTime,
-        uint32 defaultEnd
-    ) private view returns (uint32 start, uint32 end) {
-        return _dripsRange(receiver, updateTime, defaultEnd, _currTimestamp(), type(uint32).max);
+    function _dripsRangeInFuture(DripsReceiver memory receiver, uint32 defaultEnd)
+        private
+        view
+        returns (uint32 start, uint32 end)
+    {
+        return _dripsRange(receiver, defaultEnd, _currTimestamp(), type(uint32).max);
     }
 
     /// @notice Calculates the time range in which a receiver is to be dripped to.
     /// This range is capped to provide a view on drips through a specific time window.
     /// @param receiver The drips receiver
-    /// @param updateTime The time when drips are configured
     /// @param defaultEnd The end time of drips without duration
     /// @param startCap The timestamp the drips range start should be capped to
     /// @param endCap The timestamp the drips range end should be capped to
     function _dripsRange(
         DripsReceiver memory receiver,
-        uint32 updateTime,
         uint32 defaultEnd,
         uint32 startCap,
         uint32 endCap
     ) private pure returns (uint32 start, uint32 end_) {
-        start = updateTime;
+        start = startCap;
```

Comment:
@CodeSandwich  I think you are right with your example. We need actually `start=lastUpdate` here. 
However we still could remove the `if (start < updateTime) start = updateTime;` part.

I will add the testcase here for this example

Reply:
I don't think we can do that either, it covers the case of `_dripsRangeInPast` which has `startCap == 0`. T `start != 0`, then nothing would prevent `start < updateTime`.

Reply:
Why the `startCap` would be the currentTimestamp.

Reply:
Could you elaborate? `startCap` is current timestamp only in `_dripsRangeInFuture`, I've raised an example of `_dripsInPast`, which has `startCap` of `0`.

Reply:
in the `_dripsInPast` case it is now `startCap=lastUpdate`.

Reply:
Ah, right, I've missed that. Let's see what the code looks like after fixing the `currRecv.start == 0 && currRecv.duration != 0.` case. You may be right about the removal of the `if (start < updateTime) start = updateTime;` line.